front-end/src/common/stats-generators/BaseStatsGenerator.ts

```diff
@@ -1,7 +1,22 @@
-import PlayerStatModel from "../../../../../../Common/models/playerstat.model";
+import { Theme } from "@mui/material";
+import PlayerStatModel from "../../../../Common/models/playerstat.model";
 
 export abstract class BaseStatGenerator {
 
+    static getColor(stat: number, theme: Theme): string {
+        return theme.palette.text.primary;
+    }
+
+    static getColorHelper(stat: number, theme: Theme, bp1: number, bp2: number, bp3: number): string {
+        if (stat >= bp1)
+            return theme.palette.first.main;
+        else if (stat >= bp2)
+            return theme.palette.second.main;
+        else if (stat >= bp3)
+            return theme.palette.third.main;
+        return theme.palette.text.primary;
```

Comment:
Shouldn't the stat thats implementing this class know about its own baselines? I dont see why you need to pass in bp1,bp2,bp3. Imo it should use an abstract method that returns these values

Reply:
Yes, it will know its own breakpoints and pass them in via the abstract getColor function